From ralf@linux-mips.org Fri Jun 23 03:55:48 2006
Date: Fri, 23 Jun 2006 11:55:43 +0100
From: Ralf Baechle <ralf@linux-mips.org>
To: gregkh@suse.de, linux-usb-devel@lists.sourceforge.net
Subject: USB: au1xxx: compile fixes for OHCI for au1200
Message-ID: <20060623105543.GH5896@linux-mips.org>
Content-Disposition: inline

From: Domen Puncer <domen.puncer@ultra.si>

Compile fixes for au1200 ohci.

First part looks a bit hackish... but it works for me.


Signed-off-by: Domen Puncer <domen.puncer@ultra.si>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci-au1xxx.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/host/ohci-au1xxx.c
+++ gregkh-2.6/drivers/usb/host/ohci-au1xxx.c
@@ -101,9 +101,11 @@ static void au1xxx_start_ohc(struct plat
 
 #endif  /* Au1200 */
 
+#ifndef CONFIG_SOC_AU1200
 	/* wait for reset complete (read register twice; see au1500 errata) */
 	while (au_readl(USB_HOST_CONFIG),
 		!(au_readl(USB_HOST_CONFIG) & USBH_ENABLE_RD))
+#endif
 		udelay(1000);
 
 	printk(KERN_DEBUG __FILE__
@@ -157,9 +159,9 @@ static int usb_ohci_au1xxx_probe(const s
 	/* Au1200 AB USB does not support coherent memory */
 	if (!(read_c0_prid() & 0xff)) {
 		pr_info("%s: this is chip revision AB !!\n",
-			dev->dev.name);
+			dev->name);
 		pr_info("%s: update your board or re-configure the kernel\n",
-			dev->dev.name);
+			dev->name);
 		return -ENODEV;
 	}
 #endif
